alacritty/src/event.rs

```diff
@@ -972,13 +983,41 @@ impl Mouse {
     }
 }
 
+/// State of touch.
+#[derive(Debug)]
+pub struct TouchState {
+    pub start_x: f64,
+    pub start_y: f64,
+    pub x: f64,
+    pub y: f64,
+}
```

Comment:
Is it really necessary to separate this from what is already tracked in the `Mouse` struct? Just seems like a waste to me.

Reply:
I don't see a way to do without tracking `start_x` and `start_y` separately, since I see nothing analogous in `Mouse`. We could always place them in `Mouse` and call them `touch_start_x` and `touch_start_y`, but that's horrible. There are many other options (I won't make a list), but all feel similarly unnatural.

There are a few additional considerations:

- It would be nice if attempting to use the mouse and touch screen simultaneously didn't result in horrifically confusing misbehavior. This is made less likely by keeping the state info as separate as possible.
- Future features (I do intend to implement drag-to-scroll eventually, just not this month and not for this PR) are even more likely to need a dedicated `TouchState` object.

I also view `Mouse.x` and `TouchState.x` as conceptually very different, and therefore it's just "right" to keep them separate, but that's just some vague intuition that I can't possibly justify.

Reply:
> It would be nice if attempting to use the mouse and touch screen simultaneously didn't result in horrifically confusing misbehavior.

I really don't care much about this. Having the touchscreen solely as mouse emulation seems fine to me. There's absolutely no need to support both at the same time.

> I also view Mouse.x and TouchState.x as conceptually very different, and therefore it's just "right" to keep them separate, but that's just some vague intuition that I can't possibly justify.

I don't see how they're different at all. Both present the last pointer X position.

> I don't see a way to do without tracking start_x and start_y separately, since I see nothing analogous in Mouse.

There is. We have `block_hint_launcher` already which is set to `true` in `mouse_moved` whenever the mouse moved by one cell. This is used to prevent the user from opening a URL when trying to select it. Touch click recognition should behave similarly.

Though of course if we actually wanted mouse clicking to work **properly**, we shouldn't just emit a single click at some point in time. Realistically even with a touch click the press and release should be reported separately, otherwise applications will not work properly. So most of the time it really shouldn't be alacritty that determines if the mouse has moved or not, it should be the application handling the click. It's just that in some scenarios we're handling the click in Alacritty.